business/istio_validations.go

```diff
@@ -258,41 +188,79 @@ func runObjectCheckers(objectCheckers []ObjectChecker) models.IstioValidations {
 	return objectTypeValidations
 }
 
+// The following idea is used underneath: if errChan has at least one record, we'll effectively cancel the request (if scheduled in such order). On the other hand, if we can't
+// write to the buffered errChan, we just ignore the error. This is because a single error is enough to cancel the whole request.
+
 func fetch(rValue *[]kubernetes.IstioObject, namespace string, service string, fetcher func(string, string) ([]kubernetes.IstioObject, error), wg *sync.WaitGroup, errChan chan error) {
 	defer wg.Done()
-	fetched, err := fetcher(namespace, service)
-	*rValue = append(*rValue, fetched...)
-	if err != nil {
-		errChan <- err
+	if len(errChan) == 0 {
+		fetched, err := fetcher(namespace, service)
+		*rValue = append(*rValue, fetched...)
+		if err != nil {
+			select {
+			case errChan <- err:
+			default:
+			}
+		}
 	}
 }
 
 // Identical to above, but since k8s layer has both (namespace, serviceentry) and (namespace) queries, we need two different functions
 func fetchNoEntry(rValue *[]kubernetes.IstioObject, namespace string, fetcher func(string) ([]kubernetes.IstioObject, error), wg *sync.WaitGroup, errChan chan error) {
 	defer wg.Done()
-	fetched, err := fetcher(namespace)
-	*rValue = append(*rValue, fetched...)
-	if err != nil && len(errChan) == 0 {
-		errChan <- err
+	if len(errChan) == 0 {
+		fetched, err := fetcher(namespace)
+		*rValue = append(*rValue, fetched...)
+		if err != nil {
+			select {
+			case errChan <- err:
```

Comment:
Yeah, it will not block if writing to the channel fails (full / closed)

Reply:
Perhaps can we put a comment ?
If I follow the logic, we need to know that errChan is max sized to 1 to understand the logic of this check, right ?
A line explain it will help others (like me :-) ) to quickly get it.

Reply:
Isn't the comment above? ``On the other hand, if we can't write to the buffered errChan, we just ignore the error.`` I can modify that a bit to explain how channels work for this context.